### ISSUE TITLE
Specify the meaning of Return 0 (of 0/1)

### ISSUE THREAD

**Issue by esilvia** [ASPRSorg/LAS#14]
I recently learned this historic information from Lewis:

Return 0 is intentionally in the spec.  It was originally for Hardware vendors to be able to indicate that a pulse was emitted but no returns were detected (I don’t recall if we were going to do 0 of 0 or 0 of 1).  This was never implemented by anyone.   It would be incredibly valuable if it were!

Clarify the meaning of Return 0 in the LAS specification.

**Comment by csevcik01** [comment]
If there is no return and therefore return 0,  0 of 0 seems more logical to me.

Anyhow there is little to no chance we can provide this info, as it is dropped at a certain stage of the processing chain.
Thinking of mobile mapping datasets with 360° scanners a decent amount of pulses can't be assigned a return, when there is no overhead structure, thus resulting in a lot of null returns in the las file.

**Comment by esilvia** [comment]
Agreed. I'm inclined to discourage this usage, but at the very least we can formalize its meaning. Not sure how those points would get digitized with a position anyway (0 XYZ?) since nothing ever got measured. It'd just be a timestamp record, basically.

**Comment by rapidlasso** [comment]
We could allow a return number of zero to generally mean the xyz position that the pulse was fired from. Whenever we have a 0th return for a laser shot it means that we have stored one xyz sample of the optical origin's trajectory. It could possibly be stored at a lower frequency (i.e. not for every shot). When we have a 0th return for a shot that has produced zero returns (number of returns of given pulse = 0) then it would basically indicate a shot that did not produce returns. But when we have the 0th return of a shot producing 1, 2, 3, ... 15 returns then it would store a sample of the optical origin's trajectory.

**Comment by csevcik01** [comment]
I think i get the idea of what Martin is saying and i see the value, especially for mobile and terrestrial scan data, although i am a bit concerned this might blow up file size.  If we include this in the specs, we should be careful with the term 'trajectory'. I think we don't want people to mix this up with trajectory related to IMU measurements. Also the magic word for me is 'allow'. One might use a 0 return this way, but it should NOT be mandatory to include 0 returns in a LAS file.

**Comment by rapidlasso** [comment]
Certainly not mandatory!!! And if only stored at the resolution of the GPS track it won't blow up the file much. But if we add meaning to a return number of 0 and a number of pulses of 0 to mark missing pulses (with no obvious xyz on the ground) we may at least write it such that its xyz has a useful meaning and such that a return number of 0 (with the same meaning) can also be used when the number of pulses is not 0. I would probably recommend to keep these zero returns in a separate (trajectory + missing return file) because the LASzip compressor won't like frequent jumps up to the aircraft and back down.
